app/javascript/react/components/MonsterTile.js

```diff
@@ -1,8 +1,22 @@
 import React, {useState, useEffect} from "react"
 import { Link } from "react-router-dom"
+import fetchUser from "../apiClient/fetchUser"
+import _ from "lodash"
+import DeleteButton from "./DeleteButton"
 
 const MonsterTile = props => {
   const { monster } = props
+  const [role, setRole] = useState({})
+
+  useEffect(() => {
+    fetchUser().then(((role) => {
+      setRole(role)
+    }))
+  }, [])
```

Comment:
Nice set up here! React out if you're interested in refactoring this to your top level component so other pages can gain access to this info as well without duplicating!